hardware_interface/src/resource_manager.cpp

```diff
@@ -1385,7 +1452,7 @@ LoanedCommandInterface ResourceManager::claim_command_interface(const std::strin
   resource_storage_->claimed_command_interface_map_[key] = true;
   std::lock_guard<std::recursive_mutex> guard(resource_interfaces_lock_);
   return LoanedCommandInterface(
-    resource_storage_->command_interface_map_.at(key),
+    *(resource_storage_->command_interface_map_.at(key)),
     std::bind(&ResourceManager::release_command_interface, this, key));
```

Comment:
Is it better to modify the `LoanedCommandInterface` and `LoanedStateInterface` to accept the shared_ptr's in the constructor, instead of handling references?

Reply:
I would say so :)

Reply:
Done but only as a syntactical sugar thing. I've made the constructor do the dereferencing... The whole dereferencing to take a reference of a shared ptr is a bad idea and I think we should refactor that asap but would not hold this PR up. I've added deprecation notices to the old constructors to remind us of the sloppy stuff.

I took a stab at removing the reference entirely and using the shared_ptr but it would require rewriting all the controller tests which I'm not available to do at the moment. It may also need a little more thinking on the design of this API...